### ISSUE TITLE
Make attachments list public

### ISSUE THREAD

**Issue by troy379** [GetStream/stream-chat-flutter#336]
**Is your feature request related to a problem? Please describe.**
I need to have access to the selected attachments list which is private in the `MessageInputState`. 

**Describe the solution you'd like**
Please make `MessageInputState._attachments` field and `_SendingAttachment` class  public.

gz#10242

**Comment by xsahil03x** [comment]
Hey @troy379, any particular reason or use-case for which you wants those field to be public? Maybe we can suggest some better alternatives.

**Comment by troy379** [comment]
I'm extending the existing MessageInput for a fully custom UI and it works really great, but it's not enough flexibility for me to implement attachments preview

**Comment by troy379** [comment]
The input theming approach doesn't fit my needs at all, unfortunately

**Comment by imtoori** [comment]
hey @troy379 what version of the sdk are you using?

**Comment by troy379** [comment]
hey @imtoori 
the latest stable - 0.2.21

**Comment by imtoori** [comment]
fyi we released the new stable `1.5.0`

Are you planning to switch to it? you may want to test it and give some feedback

Of course we can always publish a new `0.2.x` release making `MessageInputState._attachments` and `_SendingAttachment` public

Also, it would be nice if you could send us a screenshot/design to understand what you want to customize and how so that we can make sure you can do with our default widgets!

**Comment by troy379** [comment]
ok, I'll try the latest one, thanks.
btw I checked the latest version yesterday and it was 1.4.0 beta 😄

**Comment by imtoori** [comment]
yeah we released that yesterday evening 😄

**Comment by troy379** [comment]
looks like you have some troubles with custom date headers:
- the scroll doesn't work if there's a `Expanded` widget used:
```
dateDividerBuilder: (date) {
        return Row(children: [
          Expanded(
            child: Container(
              height: 1,
              color: Colors.grey,
            ),
          ),
          Text(date.toIso8601String()),
          Expanded(
            child: Container(
              height: 1,
              color: Colors.grey,
            ),
          ),
        ]);
      },
```

- the date is always displayed on top of the list:
<img width="256" alt="Снимок экрана 2021-03-17 в 12 55 39" src="https://user-images.githubusercontent.com/12077345/111456777-140c8480-8720-11eb-89cb-7dae130b3777.png">

**Comment by troy379** [comment]
Could you please push a quick-fix for the `0.2.x`? We're about to have a new version released this week and it's dangerous to us to push such a major change untested. Thanks

**Comment by imtoori** [comment]
Released version `0.2.21+1`
I hope that works for you :)

I'll close this issue; feel free to reopen it or create another one
We'll work on making the MessageInput more and more customizable; feedback is really appreciated :)

**Comment by troy379** [comment]
It worked, thanks a lot :)
